### DISCUSSION TITLE
How to compile?

### DISCUSSION THREAD

**Discussion by Bulat3301** [apache/age#229] [Q&A]
Compile for postgres 11.16.
With:

sudo make PG_CONFIG=/usr/lib/postgresql/11/bin/pg_config install

ubuntu:/usr/lib/postgresql/11/bin$ sudo -u postgres ./psql
psql (11.16 (Ubuntu 11.16-1.pgdg20.04+1))
Type "help" for help.
postgres=# create extension age;
ERROR:  incompatible library "/usr/lib/postgresql/11/lib/age.so": version mismatch
DETAIL:  Server is version 11, library is version 12.

What's my mistake.

**Comment by pdpotter** [comment]
Did you try installing with PostgreSQL 12 before? If that is the case, please try running
```
make clean
make
```
before running
```
sudo make PG_CONFIG=/usr/lib/postgresql/11/bin/pg_config install
```
again.

**Reply by Bulat3301** [reply]
Same output.

When trying to compile with Postgres 12.11, output:
```
incubator-age-v.1.0.0-rc1$ make PG_CONFIG=/usr/lib/postgresql/12/bin/pg_config

src/backend/catalog/ag_graph.c:67:15: error: void value not ignored as it ought to be
   67 |     graph_oid = CatalogTupleInsert(ag_graph, tuple);
      |               ^
make: *** [<builtin>: src/backend/catalog/ag_graph.o] Error 1
```

**Reply by pdpotter** [reply]
I'm sorry, I wasn't clear with my instructions. Apache AGE doesn't yet support PostgreSQL 12. That is why I asked if you tried installing on PostgreSQL 12 before trying to install on PostgreSQL 11, hoping to find out what could be the root of your issue.

Please remove PostgreSQL12, install PostgreSQL 11 and then compile and install Apache AGE with:
```
make clean
make
sudo make PG_CONFIG=/usr/lib/postgresql/11/bin/pg_config install
```

**Reply by pdpotter** [reply]
If that fails as well, can you post the output of running both `pg_config` and `/usr/lib/postgresql/11/bin/pg_config`?

**Reply by Bulat3301** [reply]
Thank you.
```/usr/lib/postgresql/11/bin/pg_config``` and ```pg_config``` were  difference on output , when postgres 12 was removed everything compiled.

**Comment by muhammadshoaib** [comment]
Hello,

We are working on AGE for PG12 and hopefully a test release will be out
in the next few days.

Once done will notify you.
